restructured structure folder and added stuff

J-T-de opened the pull request:
too lazy to split the changes in separate PRs, but I think most of them are reasonable. Check out my [fork](https://github.com/J-T-de/sourcehold-maps) to see the result.

some stuff is definitely needs to be discussed
moved files around:

structure is now
- `sourcehold-maps/structure/kaitai/`
- `sourcehold-maps/structure/cheatengine/`
- `sourcehold-maps/structure/map_structure.h`

created construct definition (especially check out construct_map.py, there is lots of stuff documented which needs to be incorporated to the wiki)
- `sourcehold-maps/structure/construct`

added `README.md` for all folders I touched with brief descriptions

To discuss:
- how to document *.CSX data properly (I created a example Unit.md, but I am myself not sure if this is good)
- should we keep `map_structure.h`, it is really well documented, but so is the wiki. I also think who can understand `map_structure.h` can understand `kaitai_map.ksy`. Not sure on this one

TODO:
- finish `CONTRIBUTING.md`
- finish ` sourcehold-maps/structure/cheatengine/README.md`
- refactor `sourcehold-maps/cheatengine`, basically tear apart the scripts and put them in single lua files as I did [here](https://github.com/J-T-de/CrusaderData/tree/master/data)
- update the wiki with the findings

Gootjes commented:
Could you maybe change your setup.py lines 25-30 to make it look like this?
That will fix the build errors.

```python
    data_files={
        "sourcehold": [str(p) for p in (list(pathlib.Path().glob("*.dll")) + list(pathlib.Path().glob("*.so"))) if p.is_file()],
        "sourcehold/cheatengine": [str(p) for p in (list(pathlib.Path("cheatengine").rglob("*.CT"))) if p.is_file()],
        "sourcehold/resources": [str(p) for p in (list(pathlib.Path("resources").rglob("*.map")) + list(pathlib.Path("resources").rglob("*.sav")) + list(pathlib.Path("resources").rglob("*.msv"))) if p.is_file()],
        "sourcehold/structure": [str(p) for p in (list(pathlib.Path("structure").rglob("*"))) if p.is_file()],
        "sourcehold/examples": [str(p) for p in (list(pathlib.Path("examples").rglob("*.py"))) if p.is_file()],
        "sourcehold/tests": [str(p) for p in (list(pathlib.Path("tests").rglob("*.py"))) if p.is_file()]
    }.items(),
```

Gootjes commented:
Regarding the documentation for the .CSX files, It looks good. Let's add some stuff regarding what we found in that structure (unit location, unit path), and what we haven't found yet (worker <-> building association).

I like Kaitai because it is more dynamic. For example, the SH mission0 situation can be easily communicated in Kaitai, but is rather peculiar to do in map_structure.h. We can also just insert more pseudo-code in map_structure.h. Nobody will really mind?
For example, I would prefer to write, than to make long documentation.

Alternatively, we can make map_structure.h a dumbed down version of what is really going on, but I am not sure if that will harm more than it helps.

````
// The amount of sections specified in the directory differs for SH and SHC (...)
if Stronghold:
   MAX_SECTIONS = 100
if Stronghold Crusader:
   MAX_SECTIONS = 150

struct Directory {
 Section sections[MAX_SECTIONS]; 
}
```

J-T-de commented:
This is all I touch for now. I think the big todos are done. Can be merged.
Please notice: I touched the main README.md, restructured it a little bit and added the discord button

Still to do
- split `/cheatengine`
- wiki updates
